### PR TITLE
Implement retry support for "cvmfs_server transaction"

### PR DESCRIPTION
**Pull request by radupopescu**
Addresses [CVM-1611](https://sft.its.cern.ch/jira/browse/CVM-1611).

Allow setting the initial and max retry delays, and the maximum
number of retries.

The retry loop starts with the initial delay and doubles it at each
retry, until the max delay is reached.

### CODE REVIEW

**Comment by jblomer** [comment]
I think we should make this available to ingestion as well, perhaps in a first approach through good default values when `cvmfs_server ingest` calls `transaction`

**Comment by radupopescu** [comment]
Makes sense, but adding gateway support to the tarball ingestion workflow is to be done in [CVM-1610](https://sft.its.cern.ch/jira/browse/CVM-1610).

I quickly checked `cvmfs_server_ingest.sh`. Right now, the command line flags of that command don't conflict with `cvmfs_server transaction` so we can even expose the same flags as here.

Do the current default retry values make sense?

File: cvmfs/server/cvmfs_server_transaction.sh [hunk 1]
```diff
@@ -78,7 +95,34 @@ cvmfs_server_transaction() {
...
+        while true ; do
+          __swissknife lease -a acquire -u $repo_services_url -k $gw_key_file -p $name"/"$subpath
+          res=$?
+          if [ $res -eq 0 ] || [ $retry -eq 0 ] || [ $r -gt $num_retries ]; then
+            break
+          fi
+          echo "Retry $r/$num_retries, waiting: $delay"
+          r=$((r + 1))
+          sleep $delay
+          delay=$((delay * 2))
```
Comment (jblomer) [inline cvmfs/server/cvmfs_server_transaction.sh]: I think that needs to be `delay=$(($delay * 2))`

Comment (radupopescu) [reply cvmfs/server/cvmfs_server_transaction.sh]: I think "$" isn't needed inside double parantheses. I mean, I also checked that it works.

**Comment by jblomer** [comment]
Let's take care then of the ingest in a follow-up PR. I'd go slightly higher in the number of retries, perhaps to 100 (~8 hours).

File: cvmfs/server/cvmfs_server_transaction.sh (hunk 1 above)
Comment (jblomer) [reply cvmfs/server/cvmfs_server_transaction.sh]: Cool, I didn't know this is valid shell.

Comment (radupopescu) [reply cvmfs/server/cvmfs_server_transaction.sh]: I always have to Google parantheses, brackets, operators etc. in shell!

**Pull request merged**